src/Cake.Core/Scripting/ScriptReference.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using Cake.Core.IO;
+
+namespace Cake.Core.Scripting
+{
+    /// <summary>
+    /// Unique Script Reference
+    /// </summary>
+    public class ScriptReference
```

Comment:
Class should be sealed. Properties should have only getters, and their backing fields should be readonly.